src/app/icd/ICDManager.h

```diff
@@ -154,6 +158,8 @@ class ICDManager : public ICDListener
     // feature map that can be changed at runtime for testing purposes
     uint32_t mFeatureMap = 0;
 #endif
+
+    static ObjectPool<ICDCheckInSender, (CHIP_CONFIG_ICD_CLIENTS_SUPPORTED_PER_FABRIC * CHIP_CONFIG_MAX_FABRICS)> mICDSenderPool;
```

Comment:
Does the pool have to be static? Can it not be created at init and teardown at Shutdown?